app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,7 +67,15 @@ def pxe_with_building_hosts?(template)
       template.respond_to?(:operatingsystem_ids) &&
       template.template_kind.present? &&
       kinds.include?(template.template_kind.name) &&
-      Host.where(:build => true, :operatingsystem_id => template.operatingsystem_ids).any?
+      building_hosts(template).any?
+  end
+
+  def building_hosts(template)
+    Host.where(:build => true, :operatingsystem_id => template.operatingsystem_ids)
+  end
+
+  def building_hosts_path(template)
+    hosts_path(:search =>  "name = " << building_hosts(template).pluck(:name).join(" or ") )
```

Comment:
Instead of listing every host, you could construct a shorter search query which may help when you have *many* hosts, which would hit the URI length limit, e.g. `build = true and (os_id = 1 or os_id = 2)`